crates/wasm/src/storage.rs

```diff
@@ -9,23 +9,45 @@ use penumbra_proto::view::v1alpha1::{NotesRequest, SwapRecord};
 use penumbra_proto::DomainType;
 use penumbra_sct::Nullifier;
 use penumbra_shielded_pool::{note, Note};
+use wasm_bindgen::prelude::wasm_bindgen;
+
+#[wasm_bindgen]
+pub struct IndexedDbConstants {
+    name: String,
+    version: u32,
+
+    assets: String,
+    chain_parameters: String,
+    fmd_parameters: String,
+    last_block_synced: String,
+    notes: String,
+    spendable_notes: String,
+    swaps: String,
+    tree_commitments: String,
+    tree_hashes: String,
+    tree_last_forgotten: String,
+    tree_last_position: String,
+}
 
 pub struct IndexedDBStorage {
     db: IdbDatabase,
+    constants: IndexedDbConstants,
 }
 
 impl IndexedDBStorage {
-    pub async fn new() -> WasmResult<Self> {
-        let db_req: OpenDbRequest = IdbDatabase::open_u32("penumbra", 12)?;
+    pub async fn new(constants: IndexedDbConstants) -> WasmResult<Self> {
```

Comment:
Does wasm-pack serialize `IndexedDbConstants` properly?

Reply:
I need to add set and get functions to properly export the struct. But actually, it seems more convenient to just use `JsValue`